python/Ganga/Core/GangaRepository/SessionLock.py

```diff
@@ -443,7 +422,8 @@ def shutdown(self):
         self.locked = set()
         try:
             global session_lock_refresher
-            session_lock_refresher.stop()
+            if session_lock_refresher and not session_lock_refresher.should_stop():
+                session_lock_refresher.stop()
```

Comment:
I'd argue a boolean check here doesn't really fit that criteria given there's a clearly expected behaviour but I get the point, I'll revert